src/dataflow-types/src/types.rs

```diff
@@ -486,6 +486,34 @@ impl Arbitrary for DataflowDescription<Plan, CollectionMetadata, mz_repr::Timest
     }
 }
 
+/// Extra context to pass through when instantiating a connector for a source or
+/// sink.
+///
+/// Should be kept cheaply cloneable.
+#[derive(Debug, Clone)]
+pub struct ConnectorContext {
+    /// The level for librdkafka's logs.
+    pub librdkafka_log_level: tracing::Level,
+}
+
+impl ConnectorContext {
+    /// Constructs a connector context from a log filter describing what log
+    /// targets and levels are of interest.
+    pub fn from_log_filter(filter: &tracing_subscriber::filter::Targets) -> ConnectorContext {
+        ConnectorContext {
+            librdkafka_log_level: mz_ore::tracing::target_level(filter, "librdkafka"),
+        }
+    }
+}
```

Comment:
This new helper method centralizes the logic for extracting the librdkafka level from a `Targets`.

Reply:
looks good!